### PR TITLE
chore: type checking in one place makes bundle smaller

### PR DESCRIPTION
**Pull request by pauldambra**
#ShowerThoughts

We have ~60 instances of type checks which might be smaller in the final bundle if they all point at a function instead of doing the work...

Let's find out ......

Looks like array.js is 1% smaller

### CODE REVIEW

**Review by daibhin — APPROVED** [review]
Looks good! Is there an easy linting rule that could be added so you don't have to come back and do this again?

**Review by benjackwhite — APPROVED** [review]
I like it a lot. (Echo David's comment)

**Review by benjackwhite — APPROVED** [review]

File: eslint-rules/no-direct-boolean-check.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+module.exports = {
+    create(context) {
+        return {
+            BinaryExpression: function (node) {
+                // Check for `toString.call(obj) == '[object Boolean]'`
+                if (
+                    (node.operator === '==' || node.operator === '===') &&
```
Comment (daibhin) [inline eslint-rules/no-direct-boolean-check.js:7]: So meta 😂

**Review by daibhin — COMMENTED** [review]
Amazing!

**Pull request merged**